algorithms/C/linked-lists/merge-two-linkedlists.c

```diff
@@ -0,0 +1,130 @@
+//Merging two Linked-List
+
+#include<stdio.h>
+#include<stdlib.h>
+
+/*Structure definition:
+     data : holds the integer value of the node
+     next : holds the pointer to the next node
+*/
+struct node
+{
+    int data;
+    struct node *next;
+}*la,*lb;                         
+//We have two linked-lists: la (List A) and lb (List B)
+
+
+//Function to create List A of given size
+void lista(int size)
+{
+    int a,i;
+    struct node *temp,*newnode;
+
+    //Head node of List A
+    la=(struct node*)malloc(sizeof(struct node));
+
+    //Getting values to enter into List A
+    printf("\nEnter the value for A: ");
+    scanf("%d",&a);
+    la->data=a;
+    la->next=NULL;
+    temp=la;
+    for(i=1; i<size; i++)
+    {
+        newnode=(struct node*)malloc(sizeof(struct node));
+        printf("Enter the value for A: ");
+        scanf("%d",&a);
+        newnode->data=a;
+        newnode->next=NULL;
+        temp->next=newnode;
+        temp=newnode;
+    }
+    printf("\n");
+}
+
+//Function to create List B of given size
+void listb(int size)
+{
+    int b,i;
+    struct node *temp,*newnode;
+
+    //Head node of List B
+    lb=(struct node*)malloc(sizeof(struct node));
+    
+    //Getting values to enter into List B
+    printf("\nEnter the value for B: ");
+    scanf("%d",&b);
+    lb->data=b;
+    lb->next=NULL;
+    temp=lb;
+    for(i=1; i<size; i++)
+    {
+        newnode=(struct node*)malloc(sizeof(struct node));
+        printf("Enter the value for B: ");
+        scanf("%d",&b);
+        newnode->data=b;
+        newnode->next=NULL;
+        temp->next=newnode;
+        temp=newnode;
+    }
+    printf("\n");
+}
+
+//Function to display the Merged Linked-lists
+void display()
+{
+    struct node *temp=la;
+    while(temp!=NULL)
+    {
+        printf("%d -> ",temp->data);
+        temp=temp->next;
+    }
+    printf("\n");
+}
+
+/*Function to merge the two linked-lists:
+        List A -> List B
+ */
+void merge()
+{
+    struct node *temp=la;
+
+    //Making the last node of List A , to point to the head node of List B

```

Comment:
isn't it more like joining or concatenating the list instead of "merging" them as the order of the original lists are not reserved at the end ?

Reply:
Yes. We can say like that. Its more like concatenating the lists.